dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -330,6 +333,94 @@ public void testIsCheckInSafeWithoutRelationshipsShouldReturnTrue() {
+
+        final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
+        final Contentlet contentletSaved = esContentletAPI.checkin(contentlet, APILocator.systemUser(), false);
+        esContentletAPI.lock(contentletSaved, user, false);
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#lock(Contentlet, User, boolean)}
+     * Given Scenario: A user with {@link PermissionLevel#EDIT} permission try to lock a contentlet
+     * ExpectedResult: Should work
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void whenTryToLockShouldWork() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+
+        final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
+        final Contentlet contentletSaved = esContentletAPI.checkin(contentlet, APILocator.systemUser(), false);
+
+        addPermission(role, contentType, PermissionLevel.WRITE);
+        addPermission(role, contentletSaved, PermissionLevel.WRITE);
+
+        esContentletAPI.lock(contentletSaved, user, false);
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#lock(Contentlet, User, boolean)}
+     * Given Scenario: The contentlet's owner without permission to EDIT try to lock the contebtlet
+     * ExpectedResult: Should work
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test()
+    public void whenOwnerTryToLockShouldWork() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id())
+                 .next();
+
+        addPermission(role, contentType, PermissionLevel.WRITE);
+
+        final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
+        final Contentlet contentletSaved = esContentletAPI.checkin(contentlet, user, false);
+        esContentletAPI.lock(contentletSaved, user, false);
+    }
+
+    private void addPermission(
+            final Role role,
+            final Permissionable contentType,
+            final PermissionLevel permissionLevel)
+
+            throws DotDataException, DotSecurityException {
+
+        APILocator.getPermissionAPI().save(
+                getPermission(role, contentType, permissionLevel.getType()),
+                contentType, APILocator.systemUser(), false);
+    }
+
+    @NotNull
+    private Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5365897)